### PR TITLE
net-snmp: Simplify the build scripts

### PR DESCRIPTION
**Pull request by bvanassche**

### CODE REVIEW

**Review by asraa — CHANGES_REQUESTED** [review]
Other than that lgtm, thanks!

**Comment by asraa** [comment]
Can you repro the failure locally? Seems like you're debugging the compiler flags

**Comment by inferno-chromium** [comment]
```checking for Perl cc... x86_64-linux-gnu-gcc
checking whether x86_64-linux-gnu-gcc is a GNU C compiler... no
configure: error: This build is using a GNU C compiler (clang) while Perl has been compiled with a non-GNU (or non-working) compiler (x86_64-linux-gnu-gcc). This likely won't work for building with Perl support. Either specify a different compiler (--with-cc=PATH), disable this check (--disable-perl-cc-checks) or build without Perl (--without-perl-modules).
ERROR:root:Building fuzzers failed.
Building project net-snmp
Running command: python /home/runner/work/oss-fuzz/oss-fuzz/infra/helper.py build_fuzzers net-snmp --engine libfuzzer --sanitizer address --architecture x86_64
Failed projects: net-snmp
Error: Process completed with exit code 1.
```

**Comment by bvanassche** [comment]
> Can you repro the failure locally? Seems like you're debugging the compiler flags

I have no idea yet what causes the build failure. I will try to reproduce the build failure locally when I have the time to do so.

**Comment by bvanassche** [comment]
> ```
> checking whether x86_64-linux-gnu-gcc is a GNU C compiler... no
> configure: error: This build is using a GNU C compiler (clang) while Perl has been compiled with a non-GNU (or non-working) compiler (x86_64-linux-gnu-gcc). This likely won't work for building with Perl support. Either specify a different compiler (--with-cc=PATH), disable this check (--disable-perl-cc-checks) or build without Perl (--without-perl-modules).
> ERROR:root:Building fuzzers failed.
> Building project net-snmp
> Running command: python /home/runner/work/oss-fuzz/oss-fuzz/infra/helper.py build_fuzzers net-snmp --engine libfuzzer --sanitizer address --architecture x86_64
> Failed projects: net-snmp
> Error: Process completed with exit code 1.
> ```

Thanks, this explains at least one of the build failures. Let's try to disable the Net-SNMP Perl extensions ...

**Comment by DavidKorczynski** [comment]
We still get the same error:
```
checking if we are in the source tree so we can install Perl modules... Yes
checking for Perl cc... x86_64-linux-gnu-gcc
checking whether x86_64-linux-gnu-gcc is a GNU C compiler... no
configure: error: This build is using a GNU C compiler (clang) while Perl has been compiled with a non-GNU (or non-working) compiler (x86_64-linux-gnu-gcc). This likely won't work for building with Perl support. Either specify a different compiler (--with-cc=PATH), disable this check (--disable-perl-cc-checks) or build without Perl (--without-perl-modules).
ERROR:root:Building fuzzers failed.
```

**Comment by bvanassche** [comment]
> We still get the same error:
> 
> ```
> checking if we are in the source tree so we can install Perl modules... Yes
> checking for Perl cc... x86_64-linux-gnu-gcc
> checking whether x86_64-linux-gnu-gcc is a GNU C compiler... no
> configure: error: This build is using a GNU C compiler (clang) while Perl has been compiled with a non-GNU (or non-working) compiler (x86_64-linux-gnu-gcc). This likely won't work for building with Perl support. Either specify a different compiler (--with-cc=PATH), disable this check (--disable-perl-cc-checks) or build without Perl (--without-perl-modules).
> ERROR:root:Building fuzzers failed.
> ```

I'm considering to modify the Net-SNMP configure script such that it checks for ABI compatibility and hence accepts building Net-SNMP with clang in combination with a Perl interpreter built with gcc.

**Comment by DavidKorczynski** [comment]
> > We still get the same error:
> > ```
> > checking if we are in the source tree so we can install Perl modules... Yes
> > checking for Perl cc... x86_64-linux-gnu-gcc
> > checking whether x86_64-linux-gnu-gcc is a GNU C compiler... no
> > configure: error: This build is using a GNU C compiler (clang) while Perl has been compiled with a non-GNU (or non-working) compiler (x86_64-linux-gnu-gcc). This likely won't work for building with Perl support. Either specify a different compiler (--with-cc=PATH), disable this check (--disable-perl-cc-checks) or build without Perl (--without-perl-modules).
> > ERROR:root:Building fuzzers failed.
> > ```
> 
> I'm considering to modify the Net-SNMP configure script such that it checks for ABI compatibility and hence accepts building Net-SNMP with clang in combination with a Perl interpreter built with gcc.

Alternatively we could keep calls to `configure` and `make` here, and then call directly into `build-fuzz-tests.sh` from the build script here too - thus avoid calling `ci/build.sh`?

**Review by DavidKorczynski — APPROVED** [review]
LGTM, thanks a lot @bvanassche!

**Comment by bvanassche** [comment]
@ArashPartow , please take another look now that the build failures have been fixed.

File: projects/net-snmp/build.sh [hunk 1]
```diff
@@ -15,23 +15,8 @@
...
-  fuzzname=$(basename -- ${fuzzer%$suffix})
-  $CC $CFLAGS -c -Iinclude -Iagent/mibgroup/agentx ./testing/fuzzing/${fuzzname}_fuzzer.c -o $WORK/${fuzzname}_fuzzer.o
-  $CXX $CXXFLAGS $WORK/${fuzzname}_fuzzer.o \
-        $LIB_FUZZING_ENGINE snmplib/.libs/libnetsnmp.a \
-        agent/.libs/libnetsnmpagent.a \
-        -Wl,-Bstatic -lcrypto -Wl,-Bdynamic -lm \
-        -o $OUT/${fuzzname}_fuzzer
-done
+# Configure and build Net-SNMP and the fuzzers.
+export CC CXX CFLAGS CXXFLAGS SRC WORK OUT LIB_FUZZING_ENGINE
```
Comment (inferno-chromium) [inline projects/net-snmp/build.sh:19]: can you just remove lines 19-21

**Pull request merged**